### PR TITLE
libmachine/ssh: export the NativeClient.Session function

### PR DESCRIPTION
**Pull request by mssola**
This allows more flexibility to developers in regards to what one can
do with a SSH session. I've also added an example to showcase a situation in
which this exported function can be useful.

This commit fixes an issue that I was having in which I wanted to stream (and manipulate such stream) the stdout of the remote process.

Another solution would be to introduce a new function on the `ssh.Client` interface that returns two readers (stdout and stderr of the remote process). This would fix my situation, but it would introduce a new function on the API, and I'm not sure if that's desirable (even though this would then work for both the Native and the External clients).

Signed-off-by: Miquel Sabaté <msabate@suse.com>

### CODE REVIEW

**Comment by nathanleclaire** [comment]
> Another solution would be to introduce a new function on the ssh.Client interface that returns two readers (stdout and stderr of the remote process). This would fix my situation, but it would introduce a new function on the API, and I'm not sure if that's desirable (even though this would then work for both the Native and the External clients).

I think I'd be more interested in this solution if you're willing :)

**Comment by mssola** [comment]
@nathanleclaire sure, I'll update this patch whenever I can ;)

**Comment by mssola** [comment]
@nathanleclaire done. Read the commit message in order to understand the changes that I've done ;)

**Comment by mssola** [comment]
@nathanleclaire I've rebased to current master. Now everything should be fine

**Comment by nathanleclaire** [comment]
I think the code looks OK, be advised that we are getting close to code freeze in preparation for the 0.6.0 / Docker 1.10 release so we will probably hold off on merging this for a little while.

**Comment by mssola** [comment]
No problem, I can wait ;) Thanks for reviewing

**Comment by dgageot** [comment]
@nathanleclaire Are you LGTM for 0.7.0?

**Comment by nathanleclaire** [comment]
@dgageot Just one nit

File: libmachine/ssh/client.go [hunk 1]
```diff
@@ -312,3 +356,34 @@ func (client ExternalClient) Shell(args ...string) error {
...
+
+	log.Debug(cmd)
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, nil, err
+	}
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		_ = stdout.Close()
```
Comment (nathanleclaire) [inline libmachine/ssh/client.go]: Why swallow the errors here?  At least aggregating them with `fmt.Errorf("%s, %s", err, closeErr)` seems reasonable, no?

**Comment by dgageot** [comment]
ping @mssola Can you have this fixed? Thanks a lot!

File: libmachine/ssh/client.go (hunk 1 above)
Comment (mssola) [inline libmachine/ssh/client.go]: Yes, makes sense. I'm on it.

**Comment by mssola** [comment]
@nathanleclaire @dgageot done

**Comment by nathanleclaire** [comment]
This PR LGTM

@dgageot @jeanlaurent PTAL

**Comment by dgageot** [comment]
LGTM

**Pull request merged**